ark/utils/test_utils.py

```diff
@@ -0,0 +1,143 @@
+            io.imsave(os.path.join(base_dir, name), tif_data[i, :, :, j])
+            filelocs[fov].append(os.path.join(base_dir, fov, sub_dir, name))
+
+    return filelocs, tif_data
+
+
+def _create_multitiff(base_dir, fov_names, channel_names, shape, sub_dir, fills, dtype):
+    tif_data = _gen_tif_data(len(fov_names), len(channel_names), shape, fills, dtype)
+
+    filelocs = {}
+
+    for i, fov in enumerate(fov_names):
+        tiffpath = os.path.join(base_dir, f'{fov}.tiff')
+        io.imsave(tiffpath, tif_data[i, :, :, :], plugin='tifffile')
+        filelocs[fov] = tiffpath
+
+    return filelocs, tif_data
+
+
+def _create_mibitiff(base_dir, fov_names, channel_names, shape, sub_dir, fills, dtype):
+    tif_data = _gen_tif_data(len(fov_names), len(channel_names), shape, fills, dtype)
+
+    filelocs = {}
+
+    mass_map = tuple(enumerate(channel_names, 1))
+
+    for i, fov in enumerate(fov_names):
+        tif_obj = mi.MibiImage(tif_data[i, :, :, :],
+                               mass_map,
+                               **MIBITIFF_METADATA)
+
+        tiffpath = os.path.join(base_dir, f'{fov}.tiff')
+        tiff.write(tiffpath, tif_obj, dtype=dtype)
+        filelocs[fov] = tiffpath
+
+    return filelocs, tif_data
+
+
+TIFFMAKERS = {
+    'tiff': _create_tifs,
+    'multitiff': _create_multitiff,
+    'mibitiff': _create_mibitiff,
+}
+
+
+def create_paired_xarray_fovs(base_dir, fov_names, channel_names, img_shape=(1024, 1024),
+                              mode='tiff', delimiter=None, sub_dir=None, fills=False,
+                              dtype="int8"):
+
+    if not os.path.isdir(base_dir):
+        raise FileNotFoundError(f'{base_dir} is not a directory')
+
+    if fov_names is None or fov_names is []:
+        raise ValueError('No fov names were given...')
+
+    if channel_names is None or channel_names is []:
+        raise ValueError('No image names were given...')
+
+    if not isinstance(fov_names, list):
+        fov_names = [fov_names]
+
+    if not isinstance(channel_names, list):
+        channel_names = [channel_names]
+
+    filelocs, tif_data = TIFFMAKERS[mode](base_dir, fov_names, channel_names, img_shape, sub_dir,
+                                          fills, dtype)
+
+    if delimiter is not None:
+        fov_ids = [fov.split(delimiter)[0] for fov in fov_names]
+
+    data_xr = xr.DataArray(tif_data,
+                           coords=[fov_ids,
+                                   range(img_shape[0]),
+                                   range(img_shape[1]),
+                                   channel_names],
+                           dims=["fovs", "rows", "cols", "channels"])
+
+    return filelocs, data_xr
+
+
+def xrs_eq(a_xr, b_xr, sortdim="fovs"):
+    return b_xr.sortby(sortdim).equals(a_xr.sortby(sortdim))
+
+
+def _create_img_dir(temp_dir, fovs, imgs, img_sub_folder="TIFs", dtype="int8"):
+    tif = np.random.randint(0, 100, 1024 ** 2).reshape((1024, 1024)).astype(dtype)
```

Comment:
I think passing argument size=(1024,1024) would be faster than reshaping the vector to matrix

Reply:
this is a legacy function which is getting replaced by `_create_tifs`.  It'll get deleted soon.